pkg/operator/hostendpointscontroller/host_endpoints_controller.go

```diff
@@ -112,52 +113,51 @@ func (c *HostEndpointsController) sync() error {
 }
 
 func (c *HostEndpointsController) syncHostEndpoints() error {
+	required := hostEndpointsAsset()
+
 	discoveryDomain, err := c.getEtcdDiscoveryDomain()
 	if err != nil {
 		return fmt.Errorf("unable to determine etcd discovery domain: %v", err)
 	}
 
-	// list etcd member pods
-	pods, err := c.podLister.List(labels.Set{"k8s-app": "etcd"}.AsSelector())
+	if required.Annotations == nil {
+		required.Annotations = map[string]string{}
+	}
+	required.Annotations["alpha.installer.openshift.io/dns-suffix"] = discoveryDomain
 
-	// get dns names of ready etcd member pods
-	var addresses []string
-	for _, pod := range pods {
-		var ready bool
-		for _, condition := range pod.Status.Conditions {
-			if condition.Type == corev1.PodReady {
-				ready = condition.Status == corev1.ConditionTrue
+	// create endpoint addresses for each node
+	nodes, err := c.nodeLister.List(labels.Set{"node-role.kubernetes.io/master": ""}.AsSelector())
+	if err != nil {
+		fmt.Errorf("unable to list expected etcd member nodes: %v", err)
+	}
+	endpointAddresses := &required.Subsets[0].Addresses
```

Comment:
instead of this, create an empty one and then do an assignment at the end.  it will be easier for slow people like me.